tests/test_dag.py

```diff
@@ -623,6 +623,52 @@ def test_batch_dag_failure(self):
         with self.assertRaises(TypeError):
             node.result()
 
+    def test_dynamic_batch_dag(self):
+        d = dag.DAG(mode=Mode.BATCH, max_workers=2)
+
+        def generate_split():
+            return list(range(0, 4))
+
+        def print_result(x):
+            print(f"Result: {x}")
+            return x
+
+        split = d.submit(
+            generate_split,
+            name="split",
+            resources={"cpu": "1", "memory": "500Mi"},
+            result_format=models.ResultFormat.JSON,
+        )
+        mult_1 = d.submit_udf_stage(
+            lambda x: x * 2,
+            split,
+            expand_node_output=split,
```

Comment:
Is there a case where we'd not want the `expand_node_output` and the `split` input args to be the same? I.e could we infer the `expand_node_output` or do we always need to pass the node parameter twice?